# Identification of Xylanase Genes from the Rumen Metagenome



__Content creators:__ 

Mahdi Anvari 610700002  
Sadegh Rizi 610700007  
Amirhossein Norouzi 610700014  

**University of Tehran, Department of Biotechnology**

## Introduction

Xylanases are a group of enzymes that play a crucial role in the breakdown of hemicellulose, a major component of plant cell walls. Their ability to degrade xylan, a complex polysaccharide, into simple sugars makes them essential in various industrial applications, including the production of biofuels, paper, and food products. The rumen, a complex digestive compartment in ruminant animals, contains a diverse microbial community that produces a range of enzymes, including xylanases, to break down dietary fibers. Identifying xylanase genes from the rumen metagenome can provide valuable insights into the genetic potential of these microorganisms and enhance our understanding of their role in fiber digestion. This project aims to find and characterize xylanase genes in the rumen metagenome, investigating their diversity and potential applications in industry.

## Methods

### Step 1: Identification of Potential Xylanase Sequences

In [1]:
%use bash
cd Queries/

In [2]:
%use bash
ls -1 | wc -l

We have downloaded 1,005 known xylanase queries from the CAZy database to use in a BLAST task for comparing with our contigs.

In [3]:
%use bash
ls

Let's open one of them

In [4]:
%use bash
cat A44594.fasta

In [5]:
%use bash
cd ..
cat Queries/* > Queries.fasta

In [6]:
%use bash
head -n 20 Queries.fasta

Since these are all protein sequences, we will use the tblastn tool for the BLAST task.

In [7]:
%use bash
cd y5.final.contigs/

Now it’s time to use tblastn and create a database from our contig file.

In [ ]:
%use bash
makeblastdb -in y5.final.contigs.fa -dbtype nucl -out xylanase_db

Since this step is time-consuming and has already been completed, we will not execute this cell.

Next, we need to use this database to perform the BLAST task with our queries.

In [ ]:
%use bash
tblastn -query ../Queries.fasta -db xylanase_db -out ../tblastn_result.out -evalue 1e-5 -num_threads 7

Since this step is time-consuming and has already been completed, we will not execute this cell.

In [9]:
%use bash
cd ..

In [9]:
%use bash
head -n 20 tblastn_result.out

pir||A44594	k141_107880	48.691	191	85	5	6	187	586	26	8.89e-50	166
pir||A44594	k141_3922551	47.120	191	88	5	6	187	177	737	3.02e-47	161
pir||A44594	k141_3299547	46.316	190	89	5	7	187	779	222	1.43e-46	160
pir||A44594	k141_7053183	45.789	190	90	5	7	187	632	75	9.29e-46	158
pir||A44594	k141_3016448	47.027	185	89	4	11	188	142	690	2.06e-45	155
pir||A44594	k141_2003823	46.073	191	88	5	11	187	149	718	3.04e-45	157
pir||A44594	k141_3866072	45.026	191	92	5	6	187	121	681	1.38e-44	154
pir||A44594	k141_6831025	44.737	190	92	5	7	187	215	772	1.72e-43	152
pir||A44594	k141_6865374	47.340	188	90	4	7	187	3572	3015	1.85e-43	160
pir||A44594	k141_8364398	44.385	187	95	3	8	187	325	879	3.82e-43	152
pir||A44594	k141_405434	44.211	190	93	5	7	187	711	154	7.42e-43	150
pir||A44594	k141_2912295	44.385	187	95	4	7	186	555	1	7.55e-43	147
pir||A44594	k141_1421443	44.737	190	94	6	6	187	278	838	1.18e-42	150
pir||A44594	k141_6422758	46.524	187	91	4	8	187	472	1026	3.65e-42	155
pir||A44594	k141_8403825	49.375	160	69	4	37	187	3

In [10]:
%use bash
sed -n '$=' tblastn_result.out

381439


We obtained fewer than 400,000 results from our BLAST search. These are potential sequences coding for xylanase, but they need to be filtered first. We will perform this filtration using Python in this section. After that, we also convert the nucleotide sequences to amino acides.

### Run Filtering&Translation.ipynb

In [11]:
%use bash
grep -c '^>' filtered_output.fasta

1844


The Python notebook has filtered and translated the contigs into amino acid sequences. Now, it's time to cluster these sequences and select representatives.

### Step 2: Clustering and Selection of Representatives

In [12]:
%use bash
cd-hit -i filtered_output.fasta -o clustered_sequences.fasta -c 0.97 -n 5

Program: CD-HIT, V4.8.1 (+OpenMP), Aug 20 2021, 08:39:56
Command: cd-hit -i filtered_output.fasta -o
         clustered_sequences.fasta -c 0.97 -n 5

Started: Fri Aug 16 18:53:08 2024
                            Output                              
----------------------------------------------------------------
total seq: 1844
longest and shortest : 642 and 100
Total letters: 340001
Sequences have been sorted

Approximated minimal memory consumption:
Sequence        : 0M
Buffer          : 1 X 10M = 10M
Table           : 1 X 65M = 65M
Miscellaneous   : 0M
Total           : 76M

Table limit with the given memory limit:
Max number of representatives: 2488903
Max number of word counting entries: 90422496

comparing sequences from          0  to       1844
.
     1844  finished        583  clusters

Approximated maximum memory consumption: 77M
writing new database
writing clustering information
program completed !

Total CPU time 0.17


In [13]:
%use bash
grep -c '^>' clustered_sequences.fasta

583


We used CD-HIT to cluster the sequences, reducing our 1,844 amino acid sequences to 583 representative sequences.

### Step 3: Modeling the Conserved Region and Filtering Sequences

Our group decided to model two subfamilies of xylanase, GH10 and GH11, and search for them within our filtered data. We downloaded 30 known protein sequences for each of these subfamilies from UniProt and began the modeling process using these sequences. We used MSA and HMM for modeling these subfamilies.

In [15]:
%use bash
cd Modeling/
ls

GH10  GH11


Let's start the modeling with GH10 subfamily.

In [16]:
%use bash
cd GH10/

In [17]:
%use bash
cd Sequences/

In [18]:
%use bash
ls

C5J411.fasta.txt  O94163.fasta.txt  P40943.fasta.txt  Q5S7A8.fasta.txt
G4MLU0.fasta.txt  P07528.fasta.txt  P56588.fasta.txt  Q6PRW6.fasta.txt
I1RQU5.fasta.txt  P07529.fasta.txt  Q00177.fasta.txt  Q8J1Y4.fasta.txt
I1S3T9.fasta.txt  P23360.fasta.txt  Q01176.fasta.txt  Q96VB6.fasta.txt
O59859.fasta.txt  P23551.fasta.txt  Q0H904.fasta.txt  Q9P8J1.fasta.txt
O60206.fasta.txt  P23556.fasta.txt  Q12603.fasta.txt  W0HFK8.fasta.txt
O69231.fasta.txt  P29417.fasta.txt  Q2PGV8.fasta.txt
O74717.fasta.txt  P33559.fasta.txt  Q4JHP5.fasta.txt


These are the 30 sequences for modeling GH10 subfamily. Let's open one of them.

In [19]:
%use bash
cat C5J411.fasta.txt

>sp|C5J411|XYNC_ASPNG Probable endo-1,4-beta-xylanase C OS=Aspergillus niger OX=5061 GN=xlnC PE=2 SV=2
MVQIKVAALAMLFASQVLSEPIDPRQASVSIDTKFKAHGKKYLGNIGDQYTLTKNSKTPA
IIKADFGALTPENSMKWDATEPSRGQFSFSGSDYLVNFAQSNNKLIRGHTLVWHSQLPSW
VQSITDKNTLIEVMKNHITTVMQHYKGKIYAWDVVNEIFNEDGSLRDSVFYKVIGEDYVR
IAFETARAADPNAKLYINDYNLDSASYSKLTGMVSHVKKWIAAGIPIDGIGSQTHLSAGG
GAGISGALNALAGAGTKEIAVTELDIAGASSTDYVEVVEACLNQPKCIGITVWGVADPDS
WRSSSTPLLFDSNYNPKPAYDAIANAL


In [20]:
%use bash
cd ../

In [21]:
%use bash
cat Sequences/* > GH10_sequences.fasta

In [22]:
%use bash
head -n 20 GH10_sequences.fasta

>sp|C5J411|XYNC_ASPNG Probable endo-1,4-beta-xylanase C OS=Aspergillus niger OX=5061 GN=xlnC PE=2 SV=2
MVQIKVAALAMLFASQVLSEPIDPRQASVSIDTKFKAHGKKYLGNIGDQYTLTKNSKTPA
IIKADFGALTPENSMKWDATEPSRGQFSFSGSDYLVNFAQSNNKLIRGHTLVWHSQLPSW
VQSITDKNTLIEVMKNHITTVMQHYKGKIYAWDVVNEIFNEDGSLRDSVFYKVIGEDYVR
IAFETARAADPNAKLYINDYNLDSASYSKLTGMVSHVKKWIAAGIPIDGIGSQTHLSAGG
GAGISGALNALAGAGTKEIAVTELDIAGASSTDYVEVVEACLNQPKCIGITVWGVADPDS
WRSSSTPLLFDSNYNPKPAYDAIANAL
>sp|G4MLU0|XYN5_PYRO7 Endo-1,4-beta-xylanase 5 OS=Pyricularia oryzae (strain 70-15 / ATCC MYA-4617 / FGSC 8958) OX=242507 GN=XYL5 PE=3 SV=1
MTRLATLITLAGLLAVSPGAYAQRNRNDTGGSTGAEGLNSLAVKAGLLYFGTASDTRNFA
DEPYMSVVNNTNEFGMIVPENSMKWEATEKEPGRFSFANADRVRALTKANGQMLRCHALT
WHSQLPNFVKTTAWTRDTLTAAIESHISNEVGHFAGDCYAWDVVNEAVNENGSFRDSPFH
RTLGTDFLAISFRAAAAADPNAKLYYNDFNIETPGPKANAAMGIVRLLKEQGVRIDGVGF
QGHLTVGSTPSRAQLASQLQRFADLGVEVTYTELDIRHKSLPVSSRAAQDQARDYVSVIG
SCLDVTACVGVMVWQPTDKYSWIPETFPGTGDACLFDANMNPKPAYTSVSSLLAAAAATA
PASVVPPASVTTSKTPIQAGAGRETVSIAGLTLALSSLAFGMFML
>sp|I1RQU5|X

In [24]:
%use bash
# MSA
mafft --auto GH10_sequences.fasta > MSA_GH10_xylanases.fasta

outputhat23=16
treein = 0
compacttree = 0
stacksize: 8192 kb
rescale = 1
All-to-all alignment.
tbfast-pair (aa) Version 7.490
alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0
0 thread(s)

outputhat23=16
Loading 'hat3.seed' ... 
done.
Writing hat3 for iterative refinement
rescale = 1
Gap Penalty = -1.53, +0.00, +0.00
tbutree = 1, compacttree = 0
Constructing a UPGMA tree ... 
   20 / 30
done.

Progressive alignment ... 
STEP    26 /29 
Reallocating..done. *alloclen = 1876
STEP    29 /29 
done.
tbfast (aa) Version 7.490
alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0
1 thread(s)

minimumweight = 0.000010
autosubalignment = 0.000000
nthread = 0
randomseed = 0
blosum 62 / kimura 200
poffset = 0
niter = 16
sueff_global = 0.100000
nadd = 16
Loading 'hat3' ... done.
rescale = 1

   20 / 30
Segment   1/  1    1- 605
STEP 006-017-1  identical.   
Oscillating.

done
dvtditr (aa) Version 7.490
alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0
0 thread(s)


Stra

In [26]:
%use bash
head -n 20 MSA_GH10_xylanases.fasta

>sp|C5J411|XYNC_ASPNG Probable endo-1,4-beta-xylanase C OS=Aspergillus niger OX=5061 GN=xlnC PE=2 SV=2
----M-------VQIKVAALAMLFASQVLSEP--------------------IDPRQASV
SIDTKFKAHGKKYL--GNIGDQYTLTKNSKTPAII--KADFGALTPENSMKWDATEPSRG
--------------------------QFSFSGSDYLVNFAQSNNKLIRGHTLVWHSQLPS
WVQSIT----------------DKNTLIEVMKNHITTVM-----QHYKGKIYAWDVVNEI
FNEDGS--LR-DSVF-YKVIGEDYVRIAFETARA------ADPNAKLYINDYNLDSASYS
KLT-GMVSHVKKWIAAGIPIDGIGS--------------------QTHLSAGG-------
----GAGISGALNALAGAGTKEIAVTELDIA-------------------------GASS
TDYVEVVEACLNQPKCI-GITVWGVADPDSWRS---------------------------
--SSTP------------------------LLFDSNYNPKPAYDA---------------
----IANAL---------------------------------------------------
-
>sp|G4MLU0|XYN5_PYRO7 Endo-1,4-beta-xylanase 5 OS=Pyricularia oryzae (strain 70-15 / ATCC MYA-4617 / FGSC 8958) OX=242507 GN=XYL5 PE=3 SV=1
----M----------TRLATLITLAGLLAVSPGAYAQ----------RNRNDTGGSTGAE
GLNSLAVKAGLLYF--GTASDTRNFAD-EPYMSVVNNTNEFGMIVPENSMKWEATEKEPG
-----------------------

In [28]:
%use bash
# Modeling with HMM
hmmbuild GH10_xylanase.hmm MSA_GH10_xylanases.fasta
hmmsearch --tblout GH10_results.txt GH10_xylanase.hmm ../../clustered_sequences.fasta

# hmmbuild :: profile HMM construction from multiple sequence alignments
# HMMER 3.3.2 (Nov 2020); http://hmmer.org/
# Copyright (C) 2020 Howard Hughes Medical Institute.
# Freely distributed under the BSD open source license.
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# input alignment file:             MSA_GH10_xylanases.fasta
# output HMM file:                  GH10_xylanase.hmm
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

# idx name                  nseq  alen  mlen eff_nseq re/pos description
#---- -------------------- ----- ----- ----- -------- ------ -----------
1     MSA_GH10_xylanases      30   601   340     1.31  0.590 

# CPU time: 0.11u 0.00s 00:00:00.11 Elapsed: 00:00:00.13
# hmmsearch :: search profile(s) against a sequence database
# HMMER 3.3.2 (Nov 2020); http://hmmer.org/
# Copyright (C) 2020 Howard Hughes Medical Institute.
# Freely distributed under the BSD open source license.
# - - - - - - - - - - - - - 

    1.4e-49  164.1   0.2    1.6e-49  163.9   0.2    1.0  1  k141_770369   
    1.6e-49  163.9   0.5    6.8e-49  161.8   0.5    1.7  1  k141_9388058  
    2.2e-49  163.5   0.0      3e-49  163.0   0.0    1.1  1  k141_1004518  
    5.6e-49  162.1   0.0    6.2e-49  162.0   0.0    1.0  1  k141_1687580  
      6e-49  162.0   0.0    6.6e-49  161.9   0.0    1.0  1  k141_2899687  
      1e-48  161.3   0.2    5.1e-48  159.0   0.2    1.8  1  k141_8453879  
    1.9e-48  160.4   1.5    2.2e-48  160.2   1.5    1.0  1  k141_6119917  
    2.4e-48  160.0   0.0      3e-48  159.7   0.0    1.1  1  k141_9355598  
    1.9e-47  157.1   0.0    2.1e-47  157.0   0.0    1.0  1  k141_8034862  
    3.8e-47  156.1   0.0    4.1e-47  156.0   0.0    1.0  1  k141_5145849  
    4.7e-47  155.8   1.7    5.5e-47  155.6   1.7    1.0  1  k141_6755562  
    5.9e-47  155.5   0.4    6.5e-47  155.3   0.4    1.0  1  k141_3918953  
      7e-47  155.2   0.8    8.5e-47  155.0   0.8    1.0  1  k141_381421   
    9.6e-47  154.8   0.9 

    1.2e-30  101.8   4.9    2.7e-30  100.7   4.9    1.5  1  k141_8805185  
    1.2e-30  101.8   0.0    1.4e-30  101.7   0.0    1.0  1  k141_4835079  
    1.5e-30  101.6   0.0    1.5e-30  101.5   0.0    1.0  1  k141_8371797  
    1.5e-30  101.6   0.0      2e-30  101.1   0.0    1.1  1  k141_143768   
    1.8e-30  101.3   0.0    1.9e-30  101.2   0.0    1.0  1  k141_9041310  
    2.8e-30  100.6   0.3    6.2e-30   99.5   0.3    1.5  1  k141_7833819  
    3.2e-30  100.5   0.0    7.8e-30   99.2   0.0    1.5  1  k141_981745   
    3.7e-30  100.2   0.0    3.9e-30  100.2   0.0    1.0  1  k141_7501167  
    4.4e-30  100.0   0.1      5e-30   99.8   0.1    1.0  1  k141_2015286  
    4.4e-30  100.0   0.0    6.2e-30   99.5   0.0    1.1  1  k141_3368813  
    4.6e-30   99.9   0.0    5.2e-30   99.8   0.0    1.0  1  k141_3970024  
    5.2e-30   99.8   0.1    5.9e-30   99.6   0.1    1.0  1  k141_3873057  
    5.6e-30   99.7   0.0    6.2e-30   99.5   0.0    1.0  1  k141_7764613  
    6.1e-30   99.5   0.0 

    2.6e-19   64.6   0.0      3e-19   64.4   0.0    1.1  1  k141_715426   
    3.7e-19   64.1   0.0    4.6e-19   63.8   0.0    1.1  1  k141_7712328  
      4e-19   64.0   0.0      5e-19   63.7   0.0    1.1  1  k141_6844872  
    4.3e-19   63.9   0.0    4.7e-19   63.8   0.0    1.0  1  k141_1144944  
    4.7e-19   63.8   0.1    5.3e-19   63.6   0.1    1.0  1  k141_3956792  
    5.1e-19   63.6   0.0    5.4e-19   63.5   0.0    1.0  1  k141_9442212  
    5.2e-19   63.6   1.7    2.8e-18   61.2   1.7    1.8  1  k141_3380878  
    5.3e-19   63.6   0.1    6.5e-19   63.3   0.1    1.0  1  k141_5451066  
      6e-19   63.4   0.2    7.1e-19   63.2   0.2    1.0  1  k141_8646947  
    6.5e-19   63.3   0.0      8e-19   63.0   0.0    1.1  1  k141_5523933  
    8.3e-19   62.9   0.0      1e-18   62.7   0.0    1.1  1  k141_2683772  
    1.1e-18   62.6   0.0    1.4e-18   62.2   0.0    1.1  1  k141_8386520  
    1.3e-18   62.3   0.7    1.5e-18   62.1   0.7    1.0  1  k141_6460686  
    1.6e-18   62.0   0.8 

    2.2e-08   28.7   0.0    2.2e-08   28.7   0.0    1.0  1  k141_3860753  
    2.8e-08   28.3   0.1    3.4e-08   28.0   0.1    1.1  1  k141_6504065  
    5.6e-08   27.3   0.1    7.3e-08   27.0   0.1    1.2  1  k141_5528618  
    6.1e-08   27.2   0.0    6.6e-08   27.1   0.0    1.1  1  k141_8048993  
    1.5e-07   26.0   0.0    1.8e-07   25.7   0.0    1.1  1  k141_3253199  
    3.9e-06   21.3   0.3    5.3e-06   20.8   0.3    1.2  1  k141_3015703  
      6e-06   20.7   0.1    7.5e-06   20.3   0.1    1.1  1  k141_9286174  
    8.1e-05   16.9   0.0    0.00012   16.4   0.0    1.2  1  k141_5934387  
    0.00014   16.2   0.1    0.00018   15.8   0.1    1.2  1  k141_8733759  


Domain annotation for each sequence (and alignments):
>> k141_4174516  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  347.5   0.1  1.8e-105  2.4e-105      34


  MSA_GH10_xylanases 135 vlknhiktvvgrYkgkvyaWDVvNEilnedgs......lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvk 219
                          +k+hi+ vv+rYk+ vyaWDVvNE++++         +r+s+ +++ ge+++ +afe+a+eadpna L++NDYn  ++  k++ + +lv+
        k141_8352380  93 NMKHHIDAVVNRYKDVVYAWDVVNEAVQDSPVrngqspMRQSPMFQIAGEEFIYKAFEYAHEADPNALLFYNDYNDAEP-GKSQRIFELVQ 182
                         ***************************98543333478**************************************999.9********** PP

  MSA_GH10_xylanases 220 klleagvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDialele..............ateekleaqakdyvevvkaclev 296
                         ++++agvpidGiG+q+H+++ +p+ +e+ +a++++ s    ++itELDi++++e              +++ +   qa++y++++k++++ 
        k141_8352380 183 RMKAAGVPIDGIGMQGHYNIYSPTAEEIDAAITKYKSIVKHIHITELDIRVNTEqggqlnfsrgqgapVASWQNTLQADQYANLFKVLRKH 273
                         **************************************************888789**********98777778899************** PP

  MSA_GH10_xylanases 297 kkcv.gvtvWgvaD

  MSA_GH10_xylanases  57 skeeaiikkdfgsltpeNsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrY 147
                          +++a+ikk+f+s+t+eN MK++ +ep++g+f++e+ad+++nfa++ng klRgH l+WhsQ+ +w+  ++ +ke + +++knhi+ vv+rY
        k141_8446463   3 PEQQALIKKEFNSMTAENDMKPQPTEPKEGEFNWENADKIANFARQNGIKLRGHCLMWHSQIGEWMLGDNPTKEVFYQRMKNHIQAVVSRY 93 
                         578999************************************************************************************* PP

  MSA_GH10_xylanases 148 kgkvyaWDVvNEilnedgs....lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpidGiGsq 234
                         k+ vy WDVvNE++ +d +    +r+s  y+++g++++++af++areadp++ L++NDYn  ++ +k++ ++++vk +++agvpidGiG+q
        k141_8446463  94 KDVVYCWDVVNEAMTDDKNavdpYRQSAMYKLCGDEFIAKAFQFAREADPKVLLFYNDYNECDP-VKSQRIYNMVKAMKQAGVPIDGIGMQ 183
                         ****************99989999**************************************99.************************** PP

  MSA_GH10_xylanases 235 sHlsagapsvaelkk

 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  300.2   2.1   4.2e-91   5.7e-91      74     326 ..       1     271 []       1     271 [] 0.97

  Alignments for each domain:
  == domain 1  score: 300.2 bits;  conditional E-value: 4.2e-91
  MSA_GH10_xylanases  74 NsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrYkgkvyaWDVvNEilned 164
                         N MK+e +ep++g+f++egad+++nfa++ng klRgH l+WhsQ+  w++ ++ +ke + +++knhi+ vv+rYk+ vyaWDVvNE++ +d
        k141_2078416   1 NDMKPEPTEPRQGQFNWEGADRIANFARQNGIKLRGHCLMWHSQIGRWMTDDNPTKEVFYQRMKNHIEAVVNRYKDVVYAWDVVNEAMTDD 91 
                         99****************************************************************************************9 PP

  MSA_GH10_xylanases 165 gs....lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpidGiGsqsHlsagapsvaelkkal 251
                         ++    +r+s  y+++g++++++afe+a++adpna L++NDYn  ++ +k++ ++++vkk+++agvpi+GiG+q+

                         *************************************99.9************************************************99 PP

  MSA_GH10_xylanases 260 evaitELDialele.............ateekleaqakdyvevvkaclevkkcv.gvtvWgvaDkdsWls.eespllfdenynpKpaynai 335
                          ++itE+Di+++ e             +t+++ ++ a++y++ ++++++ k+++ +vt+W++ D+dsWl  +++pl +d +y+pK ay+ i
        k141_3321694 182 HIHITEFDIRVNEEmggglqfsregatVTDSVKQHLADQYARCFRVFRKHKDVIdCVTFWNLGDRDSWLGqNNYPLPWDVDYKPKMAYDYI 272
                         **********777689*********9999999999*********************************963789**************999 PP

  MSA_GH10_xylanases 336 vk 337
                          +
        k141_3321694 273 KD 274
                         76 PP

>> k141_1057675  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  281.6   1.6   1.9e-85   2.7e-85      72     3


  Alignments for each domain:
  == domain 1  score: 279.2 bits;  conditional E-value: 1e-84
  MSA_GH10_xylanases  66 dfgsltpeNsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrYkgkvyaWDV 156
                         +f s+t+eN MK+e +ep++g+f++egad+++nfa++ng klRgH l+WhsQ+  w++s++ +ke + +++k+hi+ vv+rYk+ vyaWDV
         k141_169673   1 EFSSMTAENDMKPEPTEPRQGQFNWEGADRIANFARQNGIKLRGHCLMWHSQIGRWMTSDNPTKEVFYQRMKSHIEAVVSRYKDVVYAWDV 91 
                         699**************************************************************************************** PP

  MSA_GH10_xylanases 157 vNEilnedgs....lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpidGiGsqsHlsagaps 243
                         vNE++ +d++    +r+sv y+++g++++++afe+a++adp+a L++NDYn  ++ +k++ ++++vkk+++agvpi+GiG+q+H+++  p+
         k141_169673  92 VNEAMTDDANaqdpYRQSVMYKLCGDEFIAKAFEYAHAADPKALLFYNDYNECDP-VKSQRIYNMVKKMKDAGVPIHGIGMQGHYNIYGPK 181
                         ******999888899****************************


  MSA_GH10_xylanases 225 gvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDialele..............ateekleaqakdyvevvkaclevkkcv. 300
                         gvpidGiG+q+H+++  p++++l+ka++++ +    ++itELD+++++e              ++      q+++y++++k++++ k+++ 
        k141_5537034 184 GVPIDGIGMQGHYNIYFPEEEQLEKAITRFKEIVNIIHITELDLRTNTEtggqlmfsrgeakpQAPYIGTLQEDQYARLFKIFRKHKDVIk 274
                         **********************************9**********777789********9975444555779999**************** PP

  MSA_GH10_xylanases 301 gvtvWgvaDkdsW 313
                         +vt+W+++DkdsW
        k141_5537034 275 NVTFWNLSDKDSW 287
                         ************* PP

>> k141_4843626  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  269.0   0.2   1.2e-81   1.7e-81      94     335 ..       1     266 [.       1     270 [. 0.95

  Alignments for each doma

                         a +q+++++++++a++kk+f+s+t+eN  K+ +++p++g ++F++ad++++f++kng k+RgH l+WhsQ  +w++++k    ++ke + e
         k141_967616   2 ALNQRNVANEEQTALVKKEFNSVTAENDWKPGELHPQEGVWDFSKADKIADFCRKNGIKMRGHCLCWHSQFADWMFTDKkgkdVKKEVFYE 92 
                         678999************************************************************************9999999****** PP

  MSA_GH10_xylanases 135 vlknhiktvvgrYkgkvyaWDVvNEilnedgs.............lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasakle 212
                         +l++hi+tvv+rYk+ vyaWDVvNE++ +dg              +r+s +++++g++++++afe+areadpn  L +NDY++ ++  k+e
         k141_967616  93 RLRDHIHTVVNRYKDVVYAWDVVNEAIADDGAprwglrpgeepspYRQSRHFKLCGDEFIAKAFEFAREADPNGLLIYNDYSTVDP-GKRE 182
                         ******************************9888888888888899**************************************99.9*** PP

  MSA_GH10_xylanases 213 gmvklvkklleagvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDial 270
                          ++++vkk+++agvpidGiG+q+H+++  p+++ l +a++++ +l   


  MSA_GH10_xylanases 247 lkkalnalaslgvevaitELDialel 272
                          +ka+n++ +l  +++itE+Di+++ 
        k141_3525868 285 FEKAINMYLELVDDIQITEFDIRINE 310
                         ***********************544 PP

>> k141_8086075  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  248.5   0.0   2.3e-75   3.1e-75     104     337 ..       1     265 [.       1     267 [. 0.96

  Alignments for each domain:
  == domain 1  score: 248.5 bits;  conditional E-value: 2.3e-75
  MSA_GH10_xylanases 104 gkklRgHtlvWhsQlPswvssik....adketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs............lresvfyrvlge 178
                         g k+RgH l+WhsQ  +w++++k    ++ke + e+l++hi+tvv+rYk+ vyaWDVvNE++ +dg             +r+s +++++g+
        k141_8086075   1 GIKMRGHCLCWHSQFADWMFTDKkgkeVKKEVFYERLRDHIHTVVNRYKDVVYAWDVVNEAMADDGGprwgrggqepspYRQSRHFKLCGD 91 
 


  MSA_GH10_xylanases 258 gvevaitELDialeleateekleaqakdyvevvkaclevkk....cv.gvtvWgvaDkdsWls.....eespllfdenynpKpaynaivka 338
                         g+ ++itELD++ + ++ ee +++  ++y+e +k++le kk    +v +vt+W++ D++sWl+     +++pllf  + ++K+ay+ +++a
        k141_1693700 188 GLKIHITELDMH-NNDPGEESMKKLGERYQEFFKIYLEAKKsgkaNVtSVTFWNLLDENSWLTgfrreQSYPLLFKGKCEAKQAYYDVLEA 277
                         ************.66666788888888888888888776542222788**************99999899*****************9986 PP

>> k141_764324  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  222.4   2.2   1.9e-67   2.6e-67     134     338 ..       2     224 ..       1     226 [. 0.96

  Alignments for each domain:
  == domain 1  score: 222.4 bits;  conditional E-value: 1.9e-67
  MSA_GH10_xylanases 134 evlknhiktvvgrYkgkvyaWDVvNEilnedgs....lresvfyrvlgedyvkiafeaareadpnakLyiNDYnle

                         +    ++ke + e+l++hi+ vv+rYk+ vyaWDVvNE++ +dg          +r+s +++++g++++++af++areadpna L++NDY+
        k141_5897125  97 SkgkpVKKEVFYERLREHIHAVVNRYKDIVYAWDVVNEAMADDGRswpgreqspYRQSRHFQLCGDEFIAKAFQFAREADPNALLFYNDYS 187
                         999999*************************************9888888888899*********************************** PP

  MSA_GH10_xylanases 204 lesasaklegmvklvkkl 221
                           ++  k+e ++++vkk+
        k141_5897125 188 CVDE-GKRERIYNMVKKM 204
                         9988.9**********98 PP

>> k141_6147756  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  218.0   0.0   4.3e-66   5.8e-66      80     267 ..       2     201 .]       1     201 [] 0.96

  Alignments for each domain:
  == domain 1  score: 218.0 bits;  conditional E-value: 4.3e-66
  MSA_GH10_xylanases  80 aiepsrgkf

                         6777888999*****************************************************************999999********** PP

  MSA_GH10_xylanases 138 nhiktvvgrYkgkvyaWDVvNEilnedgs............lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvk 216
                         +hi+tvv+rYk+ vyaWDVvNE++ +dg             +r+s +++++g++++++afe+areadpn+ L++NDY+  ++  k+e +++
        k141_2681456  93 DHIHTVVNRYKDVVYAWDVVNEAMADDGGprwgrggqqpspYRQSRHFQLCGDEFIAKAFEFAREADPNTLLFYNDYSCVDN-GKRERIYN 182
                         ***************************977777788888899************************************9977.9******* PP

  MSA_GH10_xylanases 217 lvkklleagvpidG 230
                         +vkk+++agvpidG
        k141_2681456 183 MVKKMKDAGVPIDG 196
                         *************9 PP

>> k141_3275240  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 

 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  200.4   0.3   9.5e-61   1.3e-60      34     195 ..      18     185 .]       4     185 .] 0.94

  Alignments for each domain:
  == domain 1  score: 200.4 bits;  conditional E-value: 9.5e-61
  MSA_GH10_xylanases  34 dallkaagkkyf..GtavdqkelekskeeaiikkdfgsltpeNsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswv 122
                          + lk  +k+ f  G  v+q+++++ +++a+ik++f+s+t eN MK+e +eps+g+f++++ad++++f+++ng klRgH l+WhsQ+ +w+
        k141_4551951  18 SQGLKDVYKDCFmvGVSVNQRNVTNPEQQALIKQEFNSITCENDMKPEPTEPSEGNFNWRNADRIADFCRANGIKLRGHCLMWHSQIGKWM 108
                         45789999*99999***************************************************************************** PP

  MSA_GH10_xylanases 123 ssikadketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs....lresvfyrvlgedyvkiafeaareadpna 195
                         + ++ +ke + ++++nhi+tvv+rYk+ vyaWDVvNE++ +d +    +r+sv y+++g++++++af++areadp+a
        k14


  Alignments for each domain:
  == domain 1  score: 196.8 bits;  conditional E-value: 1.2e-59
  MSA_GH10_xylanases  33 ldallkaagkkyf..GtavdqkelekskeeaiikkdfgsltpeNsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPsw 121
                         + + lk  +k+yf  G av+q+++++++++a+ik++f s+t+eN MK+e +ep++g+f++egad+++nfa++ng klRgH l+WhsQ+  w
        k141_6886796   3 MAQGLKDVYKDYFliGVAVNQRNVTNAEQQALIKREFSSMTAENDMKPEPTEPRQGQFNWEGADRIANFARQNGIKLRGHCLMWHSQIGRW 93 
                         5677899************************************************************************************ PP

  MSA_GH10_xylanases 122 vssikadketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs....lresvfyrvlgedyvkiafeaare 190
                         ++ ++ +ke + +++knhi+ vv+rYk+ vyaWDVvNE++ +d++    +r+sv y+++g++++++afe+a++
        k141_6886796  94 MTDDNPTKEVFYQRMKNHIEAVVSRYKDVVYAWDVVNEAMTDDANaedpYRQSVMYKLCGDEFIAKAFEYAHA 166
                         ******************************************99888999*********************85 PP

>> k141_5776680  

                         ******************************99.9********************************************************* PP

  MSA_GH10_xylanases 267 Di 268
                         D+
        k141_5834011 182 DL 183
                         *6 PP

>> k141_345001  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  193.3   0.2   1.4e-58   1.9e-58     102     272 ..       2     181 ..       1     184 [] 0.94

  Alignments for each domain:
  == domain 1  score: 193.3 bits;  conditional E-value: 1.4e-58
  MSA_GH10_xylanases 102 kngkklRgHtlvWhsQlPswvssik....adketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs......lresvfyrvlgedyvk 182
                         ++g klRgH l+WhsQ+ +w+ +++      ke++ +++k+hi+ vv+rYk+ vyaWDVvNE++ +         +r+s+ +++ ge+++ 
         k141_345001   2 QHGIKLRGHCLMWHSQIGTWIYQDEkgnlLPKEEFYKRMKSHIQAVVNRYKDVVYAWDVVNEAVADSPVragqs

                         ++lge+++  af++a+eadp+a Ly+NDY ++++  +++g+v+l+++l+e+g++id iG+q+H+++  p+
        k141_9437986  92 QILGEEFIPWAFQCAHEADPDAELYYNDYGMHEP-GRRDGVVRLIRQLKERGLRIDAIGMQGHMGMDYPT 160
                         ********************************99.99***************************998765 PP

>> k141_2976020  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  189.6   0.1   1.9e-57   2.6e-57      81     230 ..       1     153 []       1     153 [] 0.98

  Alignments for each domain:
  == domain 1  score: 189.6 bits;  conditional E-value: 1.9e-57
  MSA_GH10_xylanases  81 iepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs....l 167
                         +ep++g+f++++ad++++f+++ng k+RgH l+WhsQ+  w+  ++ +ke + e++++hi+ +v+rYk+ vy WDVvNE++++d +    +
        k141_2976020   1 TEPREGQFNWTNADRIADFCRA

                         + + ++vkk+++agvpi+GiG+q+H+++  ps++++ kal+ + +    ++itELDi+ ++e
        k141_2722401  94 QRIFNMVKKMKDAGVPIHGIGMQGHYNIYGPSEEDIDKALTLYKQVVSHIHITELDIRANQE 155
                         **********************************************************5544 PP

>> k141_8400597  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  177.2   0.7     1e-53   1.4e-53      66     201 ..       2     147 .]       1     147 [] 0.95

  Alignments for each domain:
  == domain 1  score: 177.2 bits;  conditional E-value: 1e-53
  MSA_GH10_xylanases  66 dfgsltpeNsMKweaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssik....adketllevlknhiktvvgrYkgkvy 152
                         +f+s+t+eN+MK++ +ep++g+f++e+ad+++nf+++ng k+RgHtl+WhsQ+ +w+ +++      ke++ + +k+hi+ vv+rYk+ vy
        k141_8400597   2 EFNSITAENAMKPQPTEPRKGEFNWEDADRIANFCRANGIKMRGHTLM

        k141_3223482  91 NIYYPDEELLDTAISRFAELVKHIHITELDLRTNTEsggqlmfargevvpQPSYIATIQEDQYARIFRVFRKHKDVIdNVTFWNLSDRDSW 181
                         ********************************888799**********9888899999********************************* PP

  MSA_GH10_xylanases 314 ls.eespllfdenynpKpaynaivk 337
                         l  ++ pl fdeny++K++++ i +
        k141_3223482 182 LGvNNHPLPFDENYKAKSSFTVIRD 206
                         986899**************99976 PP

>> k141_8814130  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  167.7   0.2   8.1e-51   1.1e-50     152     325 ..       1     202 [.       1     203 [] 0.91

  Alignments for each domain:
  == domain 1  score: 167.7 bits;  conditional E-value: 8.1e-51
  MSA_GH10_xylanases 152 yaWDVvNEilnedgs.............lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpid 229
  

                         + lk a+++yf  G av+q+++++  ++a+i ++f+s+t+eN MK++ +ep++g+f+F+ ad+++nf+++ng k+RgH l+Wh Q+ +w+ 
         k141_770369  15 QGLKDAYRDYFtiGVAVNQRNVTNPDQQALICREFNSVTAENDMKPQPTEPRQGQFDFTRADRIANFCRQNGIKMRGHCLMWHAQIGDWMY 105
                         56899************************************************************************************** PP

  MSA_GH10_xylanases 124 sik....adketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs....lresvfyrvlge 178
                         +++      k+++ +++++hi+ vv+rYk+ vy WDVvNE++ +d +    +r+sv y++ g+
         k141_770369 106 KDEqgnlLPKDEFFKRMREHIHAVVNRYKDVVYCWDVVNEAMTDDKNaedpYRQSVMYQIAGD 168
                         *998888899********************************9998888899********997 PP

>> k141_9388058  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  161.8   0.5     5e-49   6.8e-49     129    

                         lk a+k+yf  G av+q+++++  + +iikk+f+s+t+eN  K+ +i+p++g+++Fe+ad+++nf+++ng k+RgH l+WhsQ  +w++++
        k141_6119917  10 LKDAYKNYFtiGVAVNQTNVTDPAQIEIIKKQFNSVTAENDWKPGEIHPKEGEWNFEKADKIANFCRENGIKMRGHCLCWHSQFADWMFTD 100
                         6899*************************************************************************************** PP

  MSA_GH10_xylanases 126 k....adketllevlknhiktvvgrYkgkvyaWDVvNEilnedg 165
                         k    ++ke + e+l++hi+tvv+rYk+ vyaWDVvNE++ +dg
        k141_6119917 101 KkgkpVKKEVFYERLRDHIHTVVNRYKDVVYAWDVVNEAMADDG 144
                         999999***********************************998 PP

>> k141_9355598  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  159.7   0.0   2.2e-48     3e-48      46     203 ..       3     172 .]       1     172 [] 0.92

  Alignments for each do


>> k141_5473247  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  154.6   0.9   8.1e-47   1.1e-46     103     269 ..       1     178 [.       1     187 [. 0.92

  Alignments for each domain:
  == domain 1  score: 154.6 bits;  conditional E-value: 8.1e-47
  MSA_GH10_xylanases 103 ngkklRgHtlvWhsQlPswvssik.......adketllevlknhiktvv.....grYkgkvyaWDVvNEilnedgslresvfyrvlgedyv 181
                         n+ ++RgHtlvW+sQ+P+w++++        +++e ll+++++ i++v      + Y +  ya+DVvNE+  edg++r+ ++++++g+dy+
        k141_5473247   1 NNFSMRGHTLVWYSQTPEWLFHEDfdankdyVTREVLLARMESMIRQVFenlteQGYIDLFYAYDVVNEAWMEDGTMRKNHWSEIIGDDYL 91 
                         6889*****************999999999999***************977777779999******************************* PP

  MSA_GH10_xylanases 182 kiafeaareadp.nakLyiNDYnlesasaklegmvklvkklleag..vpidGiGsqsHlsagapsvaelkkalna

  MSA_GH10_xylanases  46 Gtavdqk..elekskeeaiikkdfgsltpeNsMKweaiepsrg............kfsFegadelvnfakkngkklRgHtlvWhsQlPswv 122
                         G+a+ q+  +l ++k ++i+  +f  ltpeN++K++++ + ++             ++ ++a  l +fa+kng k+ gH lvWhsQ+P+ +
        k141_6539735   2 GAAAPQYvfNLGQEKLQEIVLDHFSILTPENELKPDSVLDVQKskglakddetavAIKLNAAKPLLKFAQKNGLKVHGHVLVWHSQTPEAF 92 
                         55555552245668999********************988888888******************************************999 PP

  MSA_GH10_xylanases 123 ssik.......adketllevlknhiktvv....grYkgkvyaWDVvNEilnedgs.lr.esvfyrvlgedyvkiafeaare.adpnakLyi 199
                         +++        +++e +l +l+n+i++v+    + Y g +++WDVvNE++n+ ++ lr +s++ r++ged+v++afe+ar+ a++ + Ly+
        k141_6539735  93 FHEGydtskpfVSREIMLGRLENYIREVLtqteEEYPGVIVSWDVVNEAINDGTNwLRqDSKWVRIIGEDFVSKAFEYARKyAAEGVLLYY 183
                         98888999**99****************9888889*****************99986658*********************88899***** PP

  MSA_GH10_xylanases 200 NDYnlesasaklegm

                         8************************9988.9************************************************************ PP

  MSA_GH10_xylanases 270 lele..............ateekleaqakdyvevvkaclevkkcv.gvtvWgvaDkdsWls.eespllfdenynpKpaynaivk 337
                         ++ e              +++ +   q+++y++++k++++ ++++ +vt+W++ DkdsWl  ++ pl fdeny+pK+   ai +
        k141_2362932  92 MNNEsggqlmfsrgeakpMPAYMSTLQTDQYARLFKVFRKHADVIdNVTFWNLGDKDSWLGvNNHPLPFDENYRPKQCMRAIRD 175
                         88879***********998999999**********************************986899************9999865 PP

>> k141_3542624  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  145.9   2.5   3.5e-44   4.8e-44      48     163 ..       2     121 ..       1     125 [. 0.97

  Alignments for each domain:
  == domain 1  score: 145.9 bits;  conditional E-value: 3.5e-44
  MSA_GH

 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  141.8   0.1   6.3e-43   8.6e-43     145     272 ..       1     140 [.       1     143 [] 0.94

  Alignments for each domain:
  == domain 1  score: 141.8 bits;  conditional E-value: 6.3e-43
  MSA_GH10_xylanases 145 grYkgkvyaWDVvNEilnedgs.............lresvfyrvlgedyvkiafeaareadpnakLyiNDYnlesasaklegmvklvkkll 222
                         +rYk+ vy WDVvNE++ +dg              +r+s +++++g++++++af++areadpna L++NDY++ ++  k+e ++++vkk++
        k141_3339881   1 NRYKDVVYCWDVVNEAMADDGGfrgprrggeepspYRQSRHFKLCGDEFIAKAFQFAREADPNALLFYNDYSTVDP-GKRERIYNMVKKMK 90 
                         59******************976677777777888899************************************99.9************* PP

  MSA_GH10_xylanases 223 eagvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDialel 272
                         +agvpidGiG+q+H+++  ps+++l ka++++++l   ++itELD++++ 
        k141_3339881  91 DAGVPIDGIGMQGHYNIYFPSEEQLDKAITRFSELVKHIN

IOPub message rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_msg_rate_limit`.

Current values:
NotebookApp.iopub_msg_rate_limit=1000.0 (msgs/sec)
NotebookApp.rate_limit_window=3.0 (secs)




  MSA_GH10_xylanases 172 fyrvlged....yvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpidGiGsqsHlsagapsvaelkkalnalaslg 258
                          +r++ e+     vk  f aa+e +p+a L iND+n+ +a       ++l++ lleagvpi  +G+qsH + g +  ++l++ l+++++  
        k141_4537008  91 ITRICKEKgrvgLVKEVFAAAKESNPDAVLLINDFNTSEA------YAELIEALLEAGVPISAVGIQSHQHQGYWGLEKLNRVLERFSRFS 175
                         7777776544458************************966......899****************************************** PP

  MSA_GH10_xylanases 259 vevaitE 265
                         + ++ tE
        k141_4537008 176 LPIHFTE 182
                         ******9 PP

>> k141_417097  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   91.4   0.0   1.4e-27   1.9e-27      36     144 ..      81     212 ..      62     214 .] 0.94

  Alignments for each domain:
  == domain 1  score:

                         89****************99989999999999*************9666666666999************88654154 PP

  == domain 2  score: 36.8 bits;  conditional E-value: 5.5e-11
  MSA_GH10_xylanases 158 NEilnedgslresvfyrv.lgedyvkiafeaareadp.nakLyiNDYnlesasaklegmvklvkkllea 224
                         NE l++d++  +s + +v   e+++  af++a++++p +  Ly+NDYn  +a  k+eg+v l+k ++e+
        k141_6822803  88 NEDLSNDTHGNNSSWWHVyQSEEFIINAFKYANKYAPaDLELYYNDYNECMA-KKREGIVALLKAVKEQ 155
                         999******99998866616799***********6663799********999.9**********99875 PP

>> k141_1962144  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   88.7   0.0   8.9e-27   1.2e-26      78     265 ..      33     217 ..      30     233 .. 0.86

  Alignments for each domain:
  == domain 1  score: 88.7 bits;  conditional E-value: 8.9e-27
  MSA_GH10_xylanase

        k141_6868187 141 yDNAITRICIEKgrvgLVREVFAAAKETDPDAVLLINDFNTSEA------YAQLIEDLLEADVPISAIGIQSHQHQGYWGLEKLNTV 221
                         26666666655333348************************966......899********************99998776666655 PP

>> k141_5933602  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   85.7   0.1   7.1e-26   9.8e-26     233     335 ..       1     118 [.       1     120 [] 0.93

  Alignments for each domain:
  == domain 1  score: 85.7 bits;  conditional E-value: 7.1e-26
  MSA_GH10_xylanases 233 sqsHlsagapsvaelkkalnalaslgvevaitELDialele.............ateekleaqakdyvevvkaclevkkcv.gvtvWgvaD 309
                         +q+H+++  p+++e+ ka++ ++     +++tELDi+++ +             +++ +   q+++yv+++k++++ k+++ +vt+W+v+D
        k141_5933602   1 MQAHYNVYGPTMEEVDKAIQLYSTVVKHIHLTELDIRVNEDmggglrfrqgasqVSDWERTLQQDQYVNLFKVLRKHKDVIdCVT


  Alignments for each domain:
  == domain 1  score: 84.6 bits;  conditional E-value: 1.5e-25
  MSA_GH10_xylanases  33 ldallkaagkkyf..GtavdqkelekskeeaiikkdfgsltpeNsMKweaiepsrg......kfsFegadelvnfakkngkklRgHtlvWh 115
                          da lk a  kyf  G++ +  ++++s  + +  ++++s+  eN+ K++a+  ++g      k s +   ++ +f++kng  +RgHtlvWh
        k141_4173172  23 ADAGLKLAFGKYFrvGNIFNGMNVRNSALQGLALTNYNSIECENETKPDATLVQNGstdtniKVSLNSCASIFDFCAKNGIGVRGHTLVWH 113
                         567788888899999***********9999999*******************99998999889999************************* PP

  MSA_GH10_xylanases 116 sQlPswvssik.......adketllevlknhiktvv....grYkg.kvyaWDVvNE 159
                         sQ+P+w++++        ++ +t+ ++++++ik++     ++Y   ++ya+DV+NE
        k141_4173172 114 SQTPQWFFKEGfnnngawVNSSTMDKRMESYIKNMFnaiqTQYPTlDLYAYDVCNE 169
                         ********9999999*9999*************99866667885448********9 PP

>> k141_7444673  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  

 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   83.3   0.0   3.8e-25   5.2e-25      88     182 ..      26     133 ..      11     134 .] 0.91

  Alignments for each domain:
  == domain 1  score: 83.3 bits;  conditional E-value: 3.8e-25
  MSA_GH10_xylanases  88 fsFegadelvnfakkngkklRgHtlvWhsQlPswvssik.......adketllevlknhiktvv....grYkgkvyaWDVvNEilnedgs. 166
                          +F++a  l  fa++ g k+ gH lvWhsQ+P+ ++++        ++ke +l +l+n+i++v+    + Y g +++WDVvNE++++ ++ 
        k141_4570102  26 VHFDAAKPLLRFAQSGGLKVHGHVLVWHSQTPEAFFHEGydsakplVSKEVMLGRLENYIREVLtqteELYPGVIVSWDVVNEAIDDGTNw 116
                         6899****************************9999888899*********************9666667****************99998 PP

  MSA_GH10_xylanases 167 lr.esvfyrvlgedyvk 182
                         lr  s++y+++ged+v+
        k141_4570102 117 LRtGSPWYKTIGEDFVN 133
                         77469**********96 PP

>> k141_3272949  
   #    score  bias 


  Alignments for each domain:
  == domain 1  score: 78.9 bits;  conditional E-value: 8.6e-24
  MSA_GH10_xylanases 198 yiNDYnlesasaklegmvklvkklleagvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDialeleateekleaqakdyve 288
                         ++NDY +     k++ +++ + k l ++  +dG+G+qsHl + +p+ + ++ aln++ +lg++++itELD++ + +++ + +++ a +y+e
        k141_9056285   1 FYNDYETALD-WKRDLIIEKILKPLLEKKLVDGMGMQSHLLMDHPDPEVYSTALNMYGALGLQIHITELDMH-NADPSGDSMHRLAMRYQE 89 
                         8*****9977.89998887666666667789*****************************************.777888999999999999 PP

  MSA_GH10_xylanases 289 vvkaclevkk....cv.gvtvWgvaDkdsWls 315
                          +k++le kk    +v +vt+W+++D+dsWls
        k141_9056285  90 FFKIYLEAKKsgaaNVtSVTFWNLRDEDSWLS 121
                         99999988753333688*************97 PP

>> k141_4506069  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ---

   1 !   75.4   0.2   9.9e-23   1.3e-22     121     218 ..       3     109 .]       1     109 [] 0.89

  Alignments for each domain:
  == domain 1  score: 75.4 bits;  conditional E-value: 9.9e-23
  MSA_GH10_xylanases 121 wvssik....adketllevlknhiktvvgrYkgkvyaWDVvNEilne....dgs..lresvfyrvlgedyvkiafeaareadpnakLyiND 201
                         w+ +++      ke++ + +k+hi+ +v  +   vy W VvNE++ +    +g   lr+s  y++ ge+++ +a+e+a e dpna L++ND
        k141_2386947   3 WMYQDEkgnlLPKEEFYANMKHHIQAIVISFMYVVYCWEVVNEAVADcpvyQGRpdLRNSAMYQIAGEEFIYTALEFALESDPNALLFYND 93 
                         77777766677899999****************************88444433356*********************************** PP

  MSA_GH10_xylanases 202 Ynlesasaklegmvklv 218
                         Yn  ++ ak++ + +lv
        k141_2386947  94 YNDAEP-AKSQRIFNLV 109
                         **9999.9***999987 PP

>> k141_4192004  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- 

        k141_3018142   4 RDVTAFKGVIDMWDVINEVVIMPVFdkYDNavTRICKDLGRiRLVKEVFAAAKESDPDAVLLINDFNTSKA------YEHLIEELLEADVP 88 
                         56789*************87543221133300333444554389************************966......67889********* PP

  MSA_GH10_xylanases 228 idGiGsqsHlsagapsvaelkkalnalaslgvevaitELDia..lele................ateekleaqakdyvevvkaclevkkcv 300
                         i  iG+qsH + g +  ++l++ l+++++ g+ ++ tE  +                        t e  e+qa++ +e+ +++   + + 
        k141_3018142  89 IGAIGIQSHQHQGYWGLEKLNDVLERYSRFGLPIHFTENTLIsgDI-MpghivdlndwqvnewpSTPEGEERQAREIAEMYSVLFAHPLVE 178
                         **************************************76552322.147788899999999988899999***********999999999 PP

  MSA_GH10_xylanases 301 gvtvWgvaDkdsWlseespllfdenynpKpayna 334
                         ++t+W++ D   Wl+  s +++++n   Kp+y+a
        k141_3018142 179 AITTWDFNDG-CWLKAPSGFVHEDNT-LKPSYHA 210
                         ********95.8*********99885.6888876 PP

>> k141_5942936  
   #    

>> k141_6566158  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   71.2   0.0   1.9e-21   2.6e-21     108     200 ..       1     116 []       1     116 [] 0.83

  Alignments for each domain:
  == domain 1  score: 71.2 bits;  conditional E-value: 1.9e-21
  MSA_GH10_xylanases 108 RgHtlvWhsQlPswvssik.......adketllevlknhiktvv....grYkg.kvyaWDVvNEilnedgs.lr........esvfyrvlg 177
                         RgHt+vW+sQ+P+w+++++       ++k+ + ++l++ ik+      ++Y + +vya+DV+NE +++dg  +r         s + ++ g
        k141_6566158   1 RGHTFVWYSQTPDWFFRENfsnngayVSKDIMNKRLESMIKNTFealkTQYPNlDVYAYDVCNELFKNDGGgMRpagnagsgGSTWVQIYG 91 
                         9*************************88999999988888888744446775558************999876633333333589****** PP

  MSA_GH10_xylanases 178 ed.yvkiafeaareadpn.akLyiN 200
                         +d +v  af++ar+++p   kL

                         ****************7876 PP

>> k141_1956985  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   69.4   0.0   6.8e-21   9.3e-21      40     140 ..      11     132 .]       3     132 .] 0.92

  Alignments for each domain:
  == domain 1  score: 69.4 bits;  conditional E-value: 6.8e-21
  MSA_GH10_xylanases  40 agkkyfGtavdqkelekskeeaiikkdfgsltpeNsMKweaiepsrg..............kfsFegadelvnfakkngkklRgHtlvWhs 116
                         +gk  fG+av q+ +++sk +a++ k+f  ltpeN++K++++ + +                 +F++a  +  fak+ng k+ gH lvWhs
        k141_1956985  11 EGKFDFGAAVPQHAFMDSKLKALMLKQFSILTPENELKPDSVLDIQAskslvyntgdetavVVHFDAAKGVLSFAKANGLKVHGHVLVWHS 101
                         577788***********************************999888899***********9***************************** PP

  MSA_GH10_xylanases 117 QlPswvssik.......adketllevl


  Alignments for each domain:
  == domain 1  score: 65.7 bits;  conditional E-value: 8.7e-20
  MSA_GH10_xylanases 247 lkkalnalaslgv.evaitELDia.lele...ateekleaqakdyvevvkaclevkkcv.gvtvWgvaDkdsWlseespllfdenynpKpa 331
                         ++k+l+++ + +   v + ELD++ + +       e+  +qa  y++++++++e+++ +  vt+Wg  D++sW++e++pllf +n++pK+a
         k141_381101   2 VRKSLDMFRKIDGiKVSVSELDVQiNGISngkYDGEQEMTQAIFYARLFNLYKENADLIeRVTFWGYKDNTSWRAESAPLLFKSNLEPKEA 92 
                         67899999997555**********53333555555666778999*********************************************** PP

  MSA_GH10_xylanases 332 ynaivka 338
                         y+a++++
         k141_381101  93 YYAVLNT 99 
                         ****986 PP

>> k141_1792251  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   65.1   0.0   1.3e-19   1.8e-19     128     234 ..  


  Alignments for each domain:
  == domain 1  score: 64.4 bits;  conditional E-value: 2.2e-19
  MSA_GH10_xylanases 131 tllevlknhiktvvgrYkgkvyaWDVvNEilnedgs.lresvfyrvlge....dyvkiafeaareadpnakLyiNDYnlesasaklegmvk 216
                         +++++  + i+  v+ Ykg +  WDV+NE++        +   +r++ e      vk  f aa+e +p+a+L iND+n+        +  +
         k141_715426   5 EIMRRQLERIHREVTAYKGVINLWDVINEVVIRPVFdKYDYAVTRICKEkgrvRLVKEVFTAAKECNPEARLLINDFNTSA------DYEN 89 
                         4444445679999****************986543213333334444441113589**********************984......5789 PP

  MSA_GH10_xylanases 217 lvkklleagvpidGiGsqsHlsagapsvaelkkalnalaslg 258
                         l+++llea+vpi  +G+qsH + g + +++l++ l+++++ g
         k141_715426  90 LLEELLEADVPISAVGIQSHQHQGYWGGEKLEDVLERFSRFG 131
                         9*************************************9987 PP

>> k141_7712328  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --

        k141_5451066  10 MNYSQADKMIAWAQERGIGVRGHVLVWDAYMTPWFFHEGydeknpiADPETMRARLACYIERVIthfeKKFPGVIYCWDVVNEAIGDSAAE 100
                         589****************************9999998889999*99*****************4444455678***********998877 PP

  MSA_GH10_xylanases 168 res 170
                         +++
        k141_5451066 101 WNA 103
                         665 PP

>> k141_8646947  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   63.2   0.2   5.2e-19   7.1e-19      87     170 ..      23     118 ..       5     126 .. 0.86

  Alignments for each domain:
  == domain 1  score: 63.2 bits;  conditional E-value: 5.2e-19
  MSA_GH10_xylanases  87 kfsFegadelvnfakkngkklRgHtlvWhs..QlPswvssik..........adketllevlknhiktvvgrYkgkvyaWDVvNEilnedg 165
                         + ++++ad++ ++++++gk +RgH lvW+   Q P +++++           +++e l + + + i

   1 !   60.9   0.0   2.6e-18   3.5e-18     140     309 ..       6     191 ..       3     194 .] 0.83

  Alignments for each domain:
  == domain 1  score: 60.9 bits;  conditional E-value: 2.6e-18
  MSA_GH10_xylanases 140 iktvvgrYkgkvyaWDVvNEilnedgs.lresvfyrvlged....yvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleag 225
                         i+  v+ +k+ +  WDV+NE++        +   +r++ ++     +k+ f  a+e +p+a L +ND+n+  +         l+   l+ag
        k141_2370290   6 IDREVTGFKEVIDMWDVINEVVIMPIFdKYDNAITRICKDKgrvgLIKTVFDKAHECNPDATLLLNDFNTSIN------YEILIDGCLNAG 90 
                         7888999*************86543321344444444433222239************************966......44578999**** PP

  MSA_GH10_xylanases 226 vpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDia..lele...............ateekleaqakdyvevvkaclevkkc 299
                         vpi  iG+qsH + g + +++l++ l++++  g+ ++ tE  +     ++                t e  ++qa++ +e+ +++ e + +
        k141_2370290  91 VPISAIGIQSHQHQGYWGKEKLNEVLDRFSTFGLPIHFTENTLIsgEIMPayiedlndwqv

   1 !   59.0   0.0   9.3e-18   1.3e-17     229     322 ..      26     121 ..      19     122 .] 0.79

  Alignments for each domain:
  == domain 1  score: 59.0 bits;  conditional E-value: 9.3e-18
  MSA_GH10_xylanases 229 dGiGsqsHlsagapsvaelkkalnalaslgvevaitELDia.leleateeklea..qakdyvevvkaclevkkcvgvtvWgvaDkdsWlse 316
                          GiG+q+H+s ++ + +e+  al+ +a+   ev+itELD++ +  ++++e  +a  +++ +++++   ++  +  +vt+Wg++D++sW+++
        k141_7140457  26 GGIGMQGHISDNN-DIDEYITALRDYAAFAPEVHITELDVKcTCSNVNREYYQAvfYKELFERLIAERRNGVNLTSVTLWGLTDDNSWIRG 115
                         69********999.899************************4333444444443114455566666666666667**************** PP

  MSA_GH10_xylanases 317 espllf 322
                           pl+f
        k141_7140457 116 ADPLVF 121
                         *****9 PP

>> k141_3315131  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    -----

                         57888884...569*******************************9876555667777******************9 PP

>> k141_3884755  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   55.7   0.6   9.4e-17   1.3e-16     127     230 ..       5     128 .]       1     128 [] 0.76

  Alignments for each domain:
  == domain 1  score: 55.7 bits;  conditional E-value: 9.4e-17
  MSA_GH10_xylanases 127 adketllevlknhiktvvgrYkgk.....vyaWDVvNEilnedgs....lr..........esvfyrvlge.dyvkiafeaareadpn.ak 196
                         ++ +t+ ++++++ik++   Yk +     +ya+DV+NE++n+ +     lr          +s + rv g+  +v++af +ar+++p+  +
        k141_3884755   5 VNSATMDKRMESYIKNMFAAYKTQypqlnLYAYDVCNEVINDGTAnqggLRptngtngqngSSAWVRVYGNnSFVEKAFTYARQYAPEgCQ 95 
                         57899*************999986444448***********85444444443322222233699*****9636***********777648*

   1 !   55.1   0.0   1.5e-16     2e-16      78     204 ..      76     205 .]      73     205 .] 0.82

  Alignments for each domain:
  == domain 1  score: 55.1 bits;  conditional E-value: 1.5e-16
  MSA_GH10_xylanases  78 weaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrYkgkvyaWDVvNEilnedgs.l 167
                         w   ep++gk  F  + + +++ +++g +++gH l+Wh     w+ + + ++e l + l+  i+  v+ +kg v  WDV+NE++       
        k141_3526777  76 WGRYEPEEGKTAFVPTMAGAQWLRERGVQVKGHPLCWHTVCAPWLMQYS-NEEILRRQLE-RIRRDVTAFKGVVDLWDVINEVVIMPVFdK 164
                         77789*****************************************988.7777777775.699999***************875433213 PP

  MSA_GH10_xylanases 168 resvf...yrvlge.dyvkiafeaareadpnakLyiNDYnl 204
                          +       r +g    vk  f aa+e +p a L iND+n+
        k141_3526777 165 YDNAItriCREMGRiRLVKEVFAAAKESNPGATLLINDFNT 205
                         33333011344443379***********************6 PP

>> k141_3369516  
   #    score  bias  c-Eva

   1 !   54.0   0.0   3.1e-16   4.2e-16     256     338 ..       2      93 ..       1      95 [. 0.89

  Alignments for each domain:
  == domain 1  score: 54.0 bits;  conditional E-value: 3.1e-16
  MSA_GH10_xylanases 256 slgvevaitELDialeleateekleaqakdyvevvkaclevkk....cv.gvtvWgvaDkdsWls.....eespllfdenynpKpaynaiv 336
                         +lg+++++tELDi+ + +++e+ +++ a +y++ ++++l+ kk    ++ +vt+W++ D++sWl+     +++pllf  + ++K+ay++++
        k141_6202951   2 ELGLQIHVTELDIH-NADPSESSMHDLALRYRKFFEIYLDAKKsgkaNItSVTFWNLLDENSWLTgfrreTSYPLLFRGKCEAKEAYYEVL 91 
                         689***********.888889999************99987653333688**************999988999*****************9 PP

  MSA_GH10_xylanases 337 ka 338
                         ka
        k141_6202951  92 KA 93 
                         87 PP

>> k141_3569566  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    --


  MSA_GH10_xylanases 270 .lele...............ateekleaqakdyvevvkaclevkkcvgvtvWgvaDkdsW 313
                            ++               +t e  e+q+++++e+++++ + + + +vt W++aD  +W
        k141_3370902 103 gHIMPpeivdlndyqipewpTTPEGEERQKNEWAEMMSVLFDHPMVEAVTGWDFADG-AW 161
                         4334489999********99889999******************999********95.45 PP

>> k141_322990  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   51.3   0.0   2.2e-15   2.9e-15     152     265 ..       2     114 ..       1     123 [. 0.86

  Alignments for each domain:
  == domain 1  score: 51.3 bits;  conditional E-value: 2.2e-15
  MSA_GH10_xylanases 152 yaWDVvNEilnedgslr.esvfyrvlged....yvkiafeaareadpnakLyiNDYnlesasaklegmvklvkklleagvpidGiGsqsHl 237
                          +WDV+NE++      r +   +r+++e      +k  f +a++a+p+a+L iND+nl ++        +++ + leag+


>> k141_8171587  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   43.3   1.3   5.5e-13   7.6e-13     156     254 ..       1     130 [.       1     133 [. 0.77

  Alignments for each domain:
  == domain 1  score: 43.3 bits;  conditional E-value: 5.5e-13
  MSA_GH10_xylanases 156 VvNEilnedgs....lresvfyrvlgedyvkiafeaare.................adpna......kLyiNDYnlesasaklegmvklvk 219
                         VvNE+++   +    lr+s +yr++g+d++  af+aa++                  d++a       L++NDYn  +++ k+  ++ l +
        k141_8171587   1 VVNEAIEPADKqetgLRNSYWYRIIGDDFMYFAFKAAHDavtelsvkyagkygidaSDEKAlsairpLLFYNDYNEWQKEKKSYIIAALNR 91 
                         8****9843333446***********************9999*********999975555533333369********99888888888888 PP

  MSA_GH10_xylanases 220 klleagvp.....idGiGsqsHlsagapsvaelkkalnal 254
                         + + +g

        k141_7179537   5 AGKFDFGVAVPGHAFGQAKLKEMILQQYSIMTPENEMKPDAVLDVAAskklaeesgddtsaAVHLDAAKPLLNFAKENGLKVHGHTLLWgk 95 
                         689999***********************************998877788**********999**************************66 PP

  MSA_GH10_xylanases 115 ...hsQlPswvssik 126
                             sQ+P+ ++++ 
        k141_7179537  96 nppESQTPKAFFHEG 110
                         666689998777665 PP

>> k141_9339147  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   46.2   0.1   7.5e-14     1e-13      78     161 ..      63     144 ..      30     147 .. 0.92

  Alignments for each domain:
  == domain 1  score: 46.2 bits;  conditional E-value: 7.5e-14
  MSA_GH10_xylanases  78 weaiepsrgkfsFegadelvnfakkngkklRgHtlvWhsQlPswvssikadketllevlknhiktvvgrYkgkvyaWDVvNEil 161
                         w   ep++gk  + ++ +++++ + 

 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   43.6   0.1   4.5e-13   6.1e-13     278     337 ..       7      68 ..       1      71 [. 0.90

  Alignments for each domain:
  == domain 1  score: 43.6 bits;  conditional E-value: 4.5e-13
  MSA_GH10_xylanases 278 kleaqakdyvevvkaclevkkcv.gvtvWgvaDkdsWls.eespllfdenynpKpaynaivk 337
                          +  q+++y+++++++++ k+++ +vt+W+++D+dsWl  ++ pl fdeny++K++++ i +
        k141_3029768   7 IATIQEDQYARIFRVFRKHKEVIdNVTFWNLSDRDSWLGvNNHPLPFDENYKAKSSFTVIRD 68 
                         5677999******************************986899**************99976 PP

>> k141_3877129  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   43.4   0.0   5.3e-13   7.3e-13     181     265 ..      29     107 ..      19     112 .. 0.94

  Alignments 

   1 !   41.3   0.0   2.3e-12   3.1e-12      41     121 ..      13     102 .]       5     102 .] 0.91

  Alignments for each domain:
  == domain 1  score: 41.3 bits;  conditional E-value: 2.3e-12
  MSA_GH10_xylanases  41 gkkyfGtavdqkelekskeeaiikkdfgsltpeNsMKweaiepsrg.........kfsFegadelvnfakkngkklRgHtlvWhsQlPsw 121
                         +   fG a + ++++++    +++++f+slt  N+ K  ++ +++          + s++ ad++  +a++n+  +RgH lvW+  + +w
        k141_5131461  13 YGFMFGGAFSFSDMNNKAFIGFLARHFNSLTCCNETKAYSLLDEQRsrtsgdgmpRMSYSRADAMISWAQRNNIRVRGHVLVWDAYMTQW 102
                         55568999999999999999999***************9999998889************************************999888 PP

>> k141_5153724  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   40.8   0.0   3.3e-12   4.6e-12      78     161 ..      35     116 ..      33     120 .. 0.91

  A


  Alignments for each domain:
  == domain 1  score: 34.2 bits;  conditional E-value: 3.3e-10
  MSA_GH10_xylanases 291 kaclevkkcv.gvtvWgvaDkdsWls.eespllfdenynpKpaynaivk 337
                         +a+++ k+++ +vt+W++ D+dsWl   ++pl fd++y+pK ay+ i +
        k141_8992782   2 RAFRKHKDVIdCVTFWNLGDRDSWLGaANYPLPFDSEYKPKMAYEFIKD 50 
                         689999******************964789**************98866 PP

>> k141_4516513  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   34.0   0.1   3.9e-10   5.3e-10     299     337 ..       3      43 ..       1      45 [. 0.90

  Alignments for each domain:
  == domain 1  score: 34.0 bits;  conditional E-value: 3.9e-10
  MSA_GH10_xylanases 299 cv.gvtvWgvaDkdsWls.eespllfdenynpKpaynaivk 337
                         ++ +vt+W++ D+dsWl  ++ pl fdeny+pK+ay+ai +
        k141_4516513   3 VIdCVTFWNLGDR

                         66666644......667899**********************************************75542222.1366778889999999 PP

  MSA_GH10_xylanases 274 .ateekleaqakdyvevvkaclevkkcvgvtvWgvaD 309
                           t e  ++qa++  e+  ++ + + + ++t+W++ D
        k141_1465015  85 pSTPEGEDRQAREISEMYTILFSHPLVEAITTWDFND 121
                         8778888899999999999999888888999999988 PP

>> k141_9484671  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   31.4   0.0   2.3e-09   3.2e-09     217     309 ..       2     111 .]       1     111 [] 0.91

  Alignments for each domain:
  == domain 1  score: 31.4 bits;  conditional E-value: 2.3e-09
  MSA_GH10_xylanases 217 lvkklleagvpidGiGsqsHlsagapsvaelkkalnalaslgvevaitELDia..lele.................ateekleaqakdyve 288
                         l+++ l+agvpi  iG+q+H + g +  ++l++ l+++   g+ ++ tE  

IOPub message rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_msg_rate_limit`.

Current values:
NotebookApp.iopub_msg_rate_limit=1000.0 (msgs/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [30]:
%use bash
head -n 20 GH10_results.txt

#                                                               --- full sequence ---- --- best 1 domain ---- --- domain number estimation ----
# target name        accession  query name           accession    E-value  score  bias   E-value  score  bias   exp reg clu  ov env dom rep inc description of target
#------------------- ---------- -------------------- ---------- --------- ------ ----- --------- ------ -----   --- --- --- --- --- --- --- --- ---------------------
k141_4174516         -          MSA_GH10_xylanases   -           1.9e-105  347.8   0.1  2.4e-105  347.5   0.1   1.0   1   0   0   1   1   1   1 -
k141_8751303         -          MSA_GH10_xylanases   -           6.4e-104  342.8   0.1    9e-104  342.3   0.1   1.1   1   0   0   1   1   1   1 -
k141_2596728         -          MSA_GH10_xylanases   -           5.1e-100  329.9   3.0  5.7e-100  329.8   3.0   1.0   1   0   0   1   1   1   1 -
k141_8352380         -          MSA_GH10_xylanases   -           6.2e-100  329.7   2.8

This file contains the most likely contigs to be xylanase proteins. We will use the top three of them for the next structural analysis.

In [31]:
%use bash
cd ../

Let's do the same for GH11 subfamily

In [32]:
%use bash
cd GH11/

In [33]:
%use bash
cd Sequences/

In [34]:
%use bash
ls

A6YAP7.fasta.txt  P0CT48.fasta.txt  P55330.fasta.txt  Q4P0L3.fasta.txt
B3VSG7.fasta.txt  P18429.fasta.txt  P55331.fasta.txt  Q4WG11.fasta.txt
G0RUP7.fasta.txt  P26220.fasta.txt  P55332.fasta.txt  Q9HFA4.fasta.txt
I1RII8.fasta.txt  P33557.fasta.txt  P55333.fasta.txt  Q9HFH0.fasta.txt
I1S2K3.fasta.txt  P36217.fasta.txt  P81536.fasta.txt  V9TXH2.fasta.txt
O43097.fasta.txt  P36218.fasta.txt  Q12550.fasta.txt  W0HJ53.fasta.txt
O74716.fasta.txt  P55328.fasta.txt  Q2LMP0.fasta.txt
P09850.fasta.txt  P55329.fasta.txt  Q2PGY1.fasta.txt


These are the 30 sequences for modeling GH10 subfamily. Let's open one of them.

In [35]:
%use bash
cat A6YAP7.fasta.txt

>sp|A6YAP7|XYN1_LEUGO Endo-1,4-beta-xylanase 1 OS=Leucoagaricus gongylophorus OX=79220 GN=Xyn1 PE=1 SV=1
MVSFIFTRIILFAAAINGAVALPMNTTEPEDFSILSRSGTPSSTGYSNGYYYSWWTDGAA
QATYANGGGGQYSLNWSGNNGNLVGGKGWNPGFNGRVIQYSGTYQPNGNSYLSVYGWTLN
PLIEYYIVESYGSYNPSSAAARKGSVNCDGANYDILTTTRYNEPSINGTQTFQQFWSVRN
PKKNPGGSISGSVSTGCHFTAWGNLGMNLGSTWNYQIVATEGYQSSGFSSITVA


In [36]:
%use bash
cd ../

In [37]:
%use bash
cat Sequences/* > GH11_sequences.fasta

In [38]:
%use bash
head -n 20 GH11_sequences.fasta

>sp|A6YAP7|XYN1_LEUGO Endo-1,4-beta-xylanase 1 OS=Leucoagaricus gongylophorus OX=79220 GN=Xyn1 PE=1 SV=1
MVSFIFTRIILFAAAINGAVALPMNTTEPEDFSILSRSGTPSSTGYSNGYYYSWWTDGAA
QATYANGGGGQYSLNWSGNNGNLVGGKGWNPGFNGRVIQYSGTYQPNGNSYLSVYGWTLN
PLIEYYIVESYGSYNPSSAAARKGSVNCDGANYDILTTTRYNEPSINGTQTFQQFWSVRN
PKKNPGGSISGSVSTGCHFTAWGNLGMNLGSTWNYQIVATEGYQSSGFSSITVA
>sp|B3VSG7|XY11A_BOTFB Endo-1,4-beta-xylanase 11A OS=Botryotinia fuckeliana (strain B05.10) OX=332648 GN=xyn11A PE=1 SV=1
MVSASSLLLAASAIAGVFSAPAAAPVSENLNVLQERALTSSATGTSGGYYYSFWTDGSGG
VTYSNGDNGQYAVSWTGNKGNFVGGKGWAVGSERSISYTGSYKPNGNSYLSVYGWTTFPL
IEYYIVEDFGTYDPSSAATEIGSVTSDGSTYKILETTRTNQPSIQGTATFKQYWSVRTSK
RTSGTVTTANHFAAWKKLGLTLGSTYDYQIVAVEGYQSGSASITVS
>sp|G0RUP7|XYN2_HYPJQ Endo-1,4-beta-xylanase 2 OS=Hypocrea jecorina (strain QM6a) OX=431241 GN=xyn2 PE=1 SV=1
MVSFTSLLAGVAAISGVLAAPAAEVESVAVEKRQTIQPGTGYNNGYFYSYWNDGHGGVTY
TNGPGGQFSVNWSNSGNFVGGKGWQPGTKNKVINFSGSYNPNGNSYLSVYGWSRNPLIEY
YIVENFGTYNPSTGATKLGEVTSDGSVYDIYRTQRVNQPSIIGTATFYQYWSVRRNHRSS
GSVNTANHFNAW

In [39]:
%use bash
# MSA
mafft --auto GH11_sequences.fasta > MSA_GH11_xylanases.fasta

outputhat23=16
treein = 0
compacttree = 0
stacksize: 8192 kb
rescale = 1
All-to-all alignment.
tbfast-pair (aa) Version 7.490
alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0
0 thread(s)

outputhat23=16
Loading 'hat3.seed' ... 
done.
Writing hat3 for iterative refinement
rescale = 1
Gap Penalty = -1.53, +0.00, +0.00
tbutree = 1, compacttree = 0
Constructing a UPGMA tree ... 
   20 / 30
done.

Progressive alignment ... 
STEP    22 /29 
Reallocating..done. *alloclen = 1485
STEP    29 /29 
done.
tbfast (aa) Version 7.490
alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0
1 thread(s)

minimumweight = 0.000010
autosubalignment = 0.000000
nthread = 0
randomseed = 0
blosum 62 / kimura 200
poffset = 0
niter = 16
sueff_global = 0.100000
nadd = 16
Loading 'hat3' ... done.
rescale = 1

   20 / 30
Segment   1/  1    1- 299
STEP 007-019-1  rejected..   
Converged.

done
dvtditr (aa) Version 7.490
alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0
0 thread(s)


Strate

In [40]:
%use bash
head -n 20 MSA_GH11_xylanases.fasta

>sp|A6YAP7|XYN1_LEUGO Endo-1,4-beta-xylanase 1 OS=Leucoagaricus gongylophorus OX=79220 GN=Xyn1 PE=1 SV=1
MVSF---------------------IFTRIILFAAAING-AVALPMNT---------TEP
EDFSILSRS-GTPSSTGYS-------NGYYYSWWTDGAAQATYANGGGGQYSLNWSGN--
NGNLVGGKGWNPGFNG-RVIQY-SGTYQP--N-GNSYLSVYGWTLNPLIEYYIVESYGSY
NPS--SAAARKGSVNCDGANYDILTTTRYNEPSINGTQ-TFQQFWSVRNPKKNPGGSISG
SVSTGCHFTAWGNLGMNLGS---TWNYQIVATEGYQSSGFSSITVA---
>sp|B3VSG7|XY11A_BOTFB Endo-1,4-beta-xylanase 11A OS=Botryotinia fuckeliana (strain B05.10) OX=332648 GN=xyn11A PE=1 SV=1
MVS-----------------------ASSLLLAASAIAG-VFSAPAAA--------PVSE
NLNVLQERA-LTSSATGTS-------GGYYYSFWTDGSGGVTYSNGDNGQYAVSWTGN--
KGNFVGGKGWAVG-SE-RSISY-TGSYKP--N-GNSYLSVYGWTTFPLIEYYIVEDFGTY
DPS--SAATEIGSVTSDGSTYKILETTRTNQPSIQGTA-TFKQYWSVRTSKRT-----SG
TVTTANHFAAWKKLGLTLGS---TYDYQIVAVEGYQ-SGSASITVS---
>sp|G0RUP7|XYN2_HYPJQ Endo-1,4-beta-xylanase 2 OS=Hypocrea jecorina (strain QM6a) OX=431241 GN=xyn2 PE=1 SV=1
MVS-----------------------FTSLLAGVAAISG-VLAAPA-----------AEV
ESVAVEKRQ-TIQP

In [41]:
%use bash
# Modeling with HMM
hmmbuild GH11_xylanase.hmm MSA_GH11_xylanases.fasta
hmmsearch --tblout GH11_results.txt GH11_xylanase.hmm ../../clustered_sequences.fasta

# hmmbuild :: profile HMM construction from multiple sequence alignments
# HMMER 3.3.2 (Nov 2020); http://hmmer.org/
# Copyright (C) 2020 Howard Hughes Medical Institute.
# Freely distributed under the BSD open source license.
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# input alignment file:             MSA_GH11_xylanases.fasta
# output HMM file:                  GH11_xylanase.hmm
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

# idx name                  nseq  alen  mlen eff_nseq re/pos description
#---- -------------------- ----- ----- ----- -------- ------ -----------
1     MSA_GH11_xylanases      30   289   222     0.88  0.589 

# CPU time: 0.07u 0.00s 00:00:00.07 Elapsed: 00:00:00.09
# hmmsearch :: search profile(s) against a sequence database
# HMMER 3.3.2 (Nov 2020); http://hmmer.org/
# Copyright (C) 2020 Howard Hughes Medical Institute.
# Freely distributed under the BSD open source license.
# - - - - - - - - - - - - - 


  MSA_GH11_xylanases  87 ..gssraikysgsyspsgnsylavYGWtrnplveyYivenygtynPssgatkkGtvtsdGstYdiytstrvnqpsieGtatFtqywsvRqs 175
                           gs+  ++y  +ysp gnsy++vYGWtr+pl+eyYive +g+++P++ ++kkGtvt dG+tYdi++++r+nqps++Gt+tF qywsvRq 
        k141_9005574 110 diGSNIVLTYDVEYSPRGNSYMCVYGWTRTPLMEYYIVEGWGSWRPGADGEKKGTVTLDGNTYDIAKTMRYNQPSLDGTQTFPQYWSVRQT 200
                         225555789********************************************************************************98 PP

  MSA_GH11_xylanases 176 krt........sgtvttanhfnaWaklGlnlgtfnYqi.vategyqssgsasit 220
                           +        sg + +++hf+aW+++Gl+++   Y + +++egy+s gsa+++
        k141_9005574 201 SGSrdnvqnnmSGIIHVGKHFDAWSQKGLDMSGTLYEVsLNIEGYRSNGSANVK 254
                         655557778889********************888988689**********985 PP

>> k141_5844946  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- 

        k141_8720139  94 RPpGNDGERKGNITLNGNTYEIAKTMRYNQPSLDGTATFPQYWSIRTTSGSannqtnymKGTIDVSKHFDAWSQKGLDMSGTLYEVsLNIE 184
                         9637999***************************************87655555667889********************888988689** PP

  MSA_GH11_xylanases 210 gyqssgsasit 220
                         gy+s gsa+++
        k141_8720139 185 GYRSNGSANVK 195
                         ********985 PP

>> k141_2049582  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  155.4   7.0   2.5e-48   4.5e-47      93     220 ..       2     139 ..       1     143 [. 0.92

  Alignments for each domain:
  == domain 1  score: 155.4 bits;  conditional E-value: 2.5e-48
  MSA_GH11_xylanases  93 kysgsyspsgnsylavYGWtrnplveyYivenygtynP.ssgatkkGtvtsdGstYdiytstrvnqpsieGtatFtqywsvRqskrt.... 178
                         +y  +y+p gnsy++vYGWtrnpl+eyYive +

                         p gnsy++vYGWt++plveyYive +g+++P ++ +++kGtvt +G+tYdi +s+r+nqps+eGt+tF qywsvR  + +        +gt
        k141_1928726   1 PRGNSYMCVYGWTKSPLVEYYIVEGWGDWRPpGNDGENKGTVTLNGNTYDIRKSMRYNQPSLEGTSTFPQYWSVRLTRGSannqtnymKGT 91 
                         68***************************963799****************************************65544445567889** PP

  MSA_GH11_xylanases 182 vttanhfnaWaklGlnlgtfnYqi 205
                         +++++hf+aW+++Gl+++   Y +
        k141_1928726  92 IDVSKHFDAWSQAGLDMSGTLYEV 115
                         *****************9888876 PP

>> k141_2356616  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !  137.3   2.2   8.4e-43   1.5e-41      99     220 ..       4     122 ..       1     124 [. 0.91

  Alignments for each domain:
  == domain 1  score: 137.3 bits;  conditional E-value: 8.4e-43
  MSA_GH11

  MSA_GH11_xylanases  26 eaaelekraltssstgasngyyysfwtdgggevtytngsggeysveWensgnfvgGkGWnpgssra......ikysgsyspsgns....yl 106
                         ++++ ++++ ++ + g s  +y  ++++g+ ++t+   ++g+y+ +W+ +++f +  G++  ++++      i+   ++s +gn+    y+
        k141_6776299  33 KTTQGQNNSSVTGNVGSSPYHYEIWYQGGNNSMTF--YDNGTYKASWNGTNDFLARVGFKYNEKQTyeelgpIDAYFKWSKQGNAggynYI 121
                         23333344445555565566666677766555555..899*******************9998876222222322234444455333339* PP

  MSA_GH11_xylanases 107 avYGWtrnplveyYivenygtynPssg..atkkGtvtsdGstYdiytstrvnqpsieGtatFtqywsvRqs.krtsgtvttanhfnaWakl 194
                          +YGWt +plveyYiv+++ + +P+++  + kkG+ t dG+tY++y+++r+n+psi+G++tF q++s R+   r+ g +++++hf+ W++l
        k141_6776299 122 GIYGWTVDPLVEYYIVDDWFS-EPGANllGSKKGEFTVDGATYEVYQNMRYNAPSIKGDQTFPQFFSKRKGgARSCGHIDITAHFKKWEEL 211
                         *******************65.6776534799*************************************9637999*************** PP

  MSA_GH11_xylanases 195 Glnlg 199
     


  Alignments for each domain:
  == domain 1  score: 47.2 bits;  conditional E-value: 3.1e-15
  MSA_GH11_xylanases  38 sstgasngyyysfwt.dgggevtytngsggeysveWensgnfvgGkGWnp..........gssraikysgsyspsgnsylavYGWtrnpl 116
                         ++ g  +g+ y++w+ +g+g++++ + ++g+++ +W+n +nf +  G n           gs+  ++y  +y+p gnsy++vYGWtrnpl
         k141_124775  12 QTRGNIGGFDYEMWNqNGQGQASM-EPKAGSFTCSWSNIENFLARMGKNYdskkqnykkiGSNIVLTYDVEYTPRGNSYMCVYGWTRNPL 100
                         455666666666665156677776.88999*************977766511111111114444679**********************7 PP

>> k141_7411831  
   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc
 ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----
   1 !   29.7   0.0   6.8e-10   1.2e-08     168     202 ..       2      36 ..       1      50 [. 0.87
   2 !   12.3   2.5   0.00014    0.0026      28      81 ..     118     169 ..      77     171 .] 0.70

  A

In [42]:
%use bash
head -n 20 GH11_results.txt

#                                                               --- full sequence ---- --- best 1 domain ---- --- domain number estimation ----
# target name        accession  query name           accession    E-value  score  bias   E-value  score  bias   exp reg clu  ov env dom rep inc description of target
#------------------- ---------- -------------------- ---------- --------- ------ ----- --------- ------ -----   --- --- --- --- --- --- --- --- ---------------------
k141_2912295         -          MSA_GH11_xylanases   -            6.4e-63  207.3  13.1   7.4e-63  207.0  13.1   1.0   1   0   0   1   1   1   1 -
k141_2003823         -          MSA_GH11_xylanases   -            3.5e-60  198.3   6.0   4.8e-60  197.8   6.0   1.1   1   0   0   1   1   1   1 -
k141_9005574         -          MSA_GH11_xylanases   -            4.8e-56  184.8   9.8   1.1e-55  183.6   9.8   1.5   1   1   0   1   1   1   1 -
k141_5844946         -          MSA_GH11_xylanases   -            1.9e-55  182.8   6.0

This file contains the most likely contigs to be xylanase proteins. We will use the top three of them for the next structural analysis.

### Step 4: Predicting the 3D Structure and Structural Alignment

For the final step, we used AlphaFold3 to predict the 3D structures and PyMOL for structural alignment. Let’s explore the results of these two analyses for each of the candidate sequences.

In [44]:
%use bash
cd ../

#### For GH10 subfamily:

In [45]:
%use bash
cd GH10/Results/

In [46]:
%use bash
ls

k141_2596728  k141_4174516  k141_8751303


#### k141_2596728

Alphaphold:

![Alphafold3-k141_2596728](Modeling/GH10/Results/k141_2596728/Alphafold-k141_2596728.png)

Pymol:

![Pymol-k141_2596728](Modeling/GH10/Results/k141_2596728/Pymol-k141_2596728.png)

RMSD = 0.993  
Industrial Xylanase PDB = 1VBR

#### k141_4174516

Alphaphold:

![Alphafold-k141_4174516](Modeling/GH10/Results/k141_4174516/Alphafold-k141_4174516.png)

Pymol:

![Pymol-k141_4174516](Modeling/GH10/Results/k141_4174516/Pymol-141_4174516.png)

RMSD = 1.004  
Industrial Xylanase PDB = 1VBR

#### k141_8751303

Alphaphold:

![Alphafold-k141_8751303](Modeling/GH10/Results/k141_8751303/Alphafold-k141_8751303.png)

Pymol:

![Pymol-k141_8751303](Modeling/GH10/Results/k141_8751303/Pymol-k141_8751303.png)

RMSD = 0.946  
Industrial Xylanase PDB = 1VBR

#### For GH11 subfamily:

In [59]:
%use bash
cd ../../../GH11/Results/

In [60]:
%use bash
ls

k141_2003823  k141_2912295  k141_9005574


#### k141_2003823

Alphaphold:

![Alphafold-k141_2003823](Modeling/GH11/Results/k141_2003823/Alphafold-k141_2003823.png)

Pymol:

![Pymol-k141_2003823](Modeling/GH11/Results/k141_2003823/Pymol-k141_2003823.png)

RMSD = 0.719  
Industrial Xylanase PDB = 1XXN

#### k141_2912295

Alphaphold:

![Alphafold-k141_2912295](Modeling/GH11/Results/k141_2912295/Alphafold-k141_2912295.png)

Pymol:

![Pymol-k141_2912295](Modeling/GH11/Results/k141_2912295/Pymol-k141_2912295.png)

RMSD = 0.957  
Industrial Xylanase PDB = 1XXN

#### k141_9005574

Alphaphold:

![Alphafold-k141_9005574](Modeling/GH11/Results/k141_9005574/Alphafold-k141_9005574.png)

Pymol:

![Pymol-k141_9005574](Modeling/GH11/Results/k141_9005574/Pymol-k141_9005574.png)

RMSD = 0.816  
Industrial Xylanase PDB = 1XXN

## Results

After predicting the 3D structures of each candidate, we performed structural alignment to compare each candidate’s structure with that of an industrially known xylanase extracted from the PDB. For the GH10 subfamily, we used the 1VBR structure, and for the GH11 subfamily, we used the 1XXN structure. We used PyMOL to obtain the RMSD scores and the TM-align website to get the TM-scores for these structural alignments. Here are the final results of the project:

In [63]:
%use bash
cd ../../

In [67]:
%use bash
cat Final_results.txt

Contig	Subfamily	Industrial structure	RMSD	TM-score
k141_2596728	GH10	1VBR	0.993	0.8619
k141_4174516	GH10	1VBR	1.004	0.90622
k141_8751303	GH10	1VBR	0.946	0.90775
k141_2003823	GH11	1XXN	0.719	0.91501
k141_2912295	GH11	1XXN	0.957	0.86914
k141_9005574	GH11	1XXN	0.816	0.91279


More information and files are available in the "Results" folder for each subfamily.

## Conclusion

As the results show, we successfully modeled two subfamilies of xylanase and identified several potential contigs encoding these enzymes in the rumen of ruminant animals. The structural analysis revealed that our candidate sequences are highly aligned with known industrial xylanases, indicating that these enzymes are indeed encoded within the rumen of ruminants.  

This finding suggests that the rumen could be an excellent resource for discovering a diverse array of industrial enzymes. These enzymes have significant potential for further scientific research and industrial applications. Exploring the rumen metagenome may therefore provide valuable insights and lead to new developments in biotechnology.

## References

https://chatgpt.com/  
http://www.cazy.org/  
https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=tblastn&PAGE_TYPE=BlastSearch&LINK_LOC=blasthome  
https://www.bioinformatics.org/cd-hit/   
https://www.uniprot.org/  
https://mafft.cbrc.jp/alignment/software/  
http://hmmer.org/  
https://alphafoldserver.com/  
https://zhanggroup.org/TM-align/  
https://pymol.org/  
https://www.rcsb.org/